cdc/model/changefeed.go

```diff
@@ -349,39 +352,36 @@ func (info *ChangeFeedInfo) fixSinkProtocol() {
 		// which is easier to troubleshoot than reporting the error directly in the bootstrap process.
 		return
 	}
-	rawQuery := sinkURIParsed.Query()
-	protocolStr := rawQuery.Get(config.ProtocolKey)
-
-	fixSinkURI := func(newProtocolStr string) {
-		oldRawQuery := sinkURIParsed.RawQuery
-		newRawQuery := rawQuery.Encode()
-		log.Info("handle incompatible protocol from sink URI",
-			zap.String("oldUriQuery", oldRawQuery),
-			zap.String("fixedUriQuery", newRawQuery))
-
-		sinkURIParsed.RawQuery = newRawQuery
-		fixedSinkURI := sinkURIParsed.String()
-		info.SinkURI = fixedSinkURI
-		info.Config.Sink.Protocol = newProtocolStr
-	}
-
-	// fix mysql sink
-	scheme := sinkURIParsed.Scheme
-	if !sink.IsMQScheme(scheme) {
-		if protocolStr != "" || info.Config.Sink.Protocol != "" {
-			maskedSinkURI, _ := util.MaskSinkURI(info.SinkURI)
-			log.Warn("sink URI or sink config contains protocol, but scheme is not mq",
-				zap.String("sinkURI", maskedSinkURI),
-				zap.String("protocol", protocolStr),
-				zap.Any("sinkConfig", info.Config.Sink))
-			// always set protocol of mysql sink to ""
-			rawQuery.Del(config.ProtocolKey)
-			fixSinkURI("")
-		}
+
+	if sink.IsMQScheme(uri.Scheme) {
```

Comment:
What happens if we got an s3 sink? Maybe we can improve it later.

Reply:
yep